plugins/inputs/sql/sql.go

```diff
@@ -413,6 +413,12 @@ func (s *SQL) ping() error {
 }
 
 func (s *SQL) prepareStatements() {
+	// We don't support prepare on ClickHouse as this requires a non-standard
+	// exec sequence on query later.
+	if s.driverName == "clickhouse" {
+		return
+	}
```

Comment:
Did this work wtih the v1 library?

Reply:
I think so as the test did not complain. We can work on doing it (we already do have a special handling on the sql output plugin). The biggest issue is that `Query()` and `Exec()` will return two completely different data-types and we would need to implement this for only ClickHouse... Would you prefer we are doing this?

Reply:
> Would you prefer we are doing this?

I'm not sure. From a let's not break anything that already works standpoint, then yes, but that seems quite involved. Are any of the other clickhouse libraries support this better out of the box?

Reply:
Thanks - I agree on holding.